src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/tools/JMap.java

```diff
@@ -50,13 +50,16 @@ public String getName() {
     }
 
     protected String getCommandFlags() {
-        return "-heap|-heap:format=b|-histo|-clstats|-finalizerinfo";
+        return "-heap|-heap:format=b[,gz=<1-9>]|-histo|-clstats|-finalizerinfo";
     }
 
     protected void printFlagsUsage() {
         System.out.println("    <no option>\tto print same info as Solaris pmap");
         System.out.println("    -heap\tto print java heap summary");
-        System.out.println("    -heap:format=b\tto dump java heap in hprof binary format");
+        System.out.println("    -heap:format=b[,gz=<1-9>]\tto dump java heap in hprof binary format");
+        System.out.println("                             \tif gz specified, the heap dump is written");
+        System.out.println("                             \tin gzipped format using the given compression level");
+        System.err.println("                             \t1 (recommended) is the fastest, 9 the strongest compression.");
```

Comment:
You need to address the fact that we now have help text that is multiple sentences. I don't like that there are no periods in this case (except you added one). And using periods implies that you should also start the sentence with an upper case letter. And if you are to do all this for this one option, then it should be done for all of them.

Reply:
Also, it looks like "-heap:format=x" is valid, and results in a dump mode of MODE_HEAP_GRAPH_GXL instead of MODE_HEAP_GRAPH_HPROF_BIN. I assume that MODE_HEAP_GRAPH_GXL and gz or not compatible. This should be made more clear in the help text. Possibly there should just be a 3rd "-heap" help entry, this one for "-help:format=x".